### DISCUSSION TITLE
A Maven archetype to remotely build, run and debug a PI4J-V2 program on a target RPI

### DISCUSSION THREAD

**Discussion by lburaggi** [Pi4J/pi4j#69] [Show and tell]
Hi,
Even if P3 or P4 boards have memory and CPU power enough to run a modern IDE like Eclipse or Netbeans, the same is not true for less equipped boards. Moreover, I would like to develop programs on a more powerful and confortable PC workstation (Linux, Windows or MacOs).
I wrote this Maven archetype that can generate PI4J-V2 project templates with remote build, run and debug capability.
A template project generated with this archetype can:
- Handle a connection to many RPI targets, configured in the pom.xml file
- Compile your code, packing the compiled code in a jar, uploading it and all the other required jars on the selected remote target
- Run the program on the remote target
- Open a remote debugging session

The archetype is IDE agnostic (I tested it using Netbeans) and depends only on maven and ant. I hope it can be useful to other people. Comments and suggestions are welcome.
[raspimaven-archetype-0.0.1-SNAPSHOT.zip](https://github.com/Pi4J/pi4j-v2/files/6006810/raspimaven-archetype-0.0.1-SNAPSHOT.zip)
To use the archetype just unzip it in a folder and read the README.md for instructions

**Comment by FDelporte** [comment]
this looks really amazing and a great contribution to the project! if I create a new repository in the the Pi4J project, would you be willing to share your work in their? and describe the use of it on a new page in https://v2.pi4j.com/getting-started/ ?

**Reply by lburaggi** [reply]
Yes, I'll do.

**Reply by FDelporte** [reply]
@lburaggi repo created and invite sent to directly work in it https://github.com/Pi4J/pi4j-maven-archetype.
for the content page for the v2-website, can you make a pull request?

**Reply by lburaggi** [reply]
I just commited the version 1.0.0-SNAPSHOT of the PI4J-v2 Maven archetype on https://github.com/Pi4J/pi4j-maven-archetype. I enriched the Readme.md file adding a more complete usage guide.

**Comment by FDelporte** [comment]
Maven archetype is available in https://github.com/Pi4J/pi4j-maven-archetype

_Trying to get this discussion marked as answered... ;-)_

**Comment by Vidhuran** [comment]
Great work 😄  Made it super easy to get a PI4J-v2 project up and running in IntelliJ with remote run and debug possibilities.
